### PR TITLE
✨ Get instances and tasks by identity

### PR DESCRIPTION
**Pull request by ElRaptorus**
**Changes:**

1. Add `flowNodeInstanceId` to all `BpmnMessages`
2. Remove unused `ProcessEndedMessage` and `ProcessTerminatedMessage`
    - These types of notifications use the `EndEventReached` and `TerminateEndEventReached` message types, located under the `BpmnMessage` namespace
3. Refactor `EventAggregatorSettings`:
    - Merge `routePaths` and `messagePaths` into `messagePaths`.
    - Rename `routeParams` to `messageParams`.
    - Split messages into `internal` and `public` message types to differentiate between messages passed inside the process engine and messages that get send to a public subscriber.
    - Public messages will not contain the identity of the ProcessInstanceOwner, but the internal messages will.
4. Add `processInstanceOwner` to datamodels for `ManualTask` and `UserTask`
5. Add `ProcessInstance` type for describing a single process instance
6. Add `IConsumerApiHttpController` interface for use with HttpControllers
7. Add functions to the `IConsumerApi` interfaces to cover the following UseCases:
    - Request a users active ProcessInstances
    - Request a users waiting ManualTasks
    - Request a users waiting UserTasks
    - Subscribe for identity-specific notifications about a new UserTask
    - Subscribe for identity-specific notifications about a finished UserTask
    - Subscribe for identity-specific notifications about a new ManualTask
    - Subscribe for identity-specific notifications about a finished ManualTask

**Issues:**

Part of https://github.com/process-engine/process_engine_runtime/issues/198

PR: #50

## How can others test the changes?

Use the upgraded Contracts for handling UserTasks and ManualTasks and try out the new UseCases.

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

### CODE REVIEW

File: typescript/src/data_models/manual_task/manual_task.ts [hunk 1]
```diff
@@ -30,4 +30,8 @@ export class ManualTask {
    * The token payload the ManualTask got suspended with.
    */
   public tokenPayload: any;
+  /**
+   * Contains the identity the UserTask belongs to.
+   */
+  public identity: any;
```
Comment (roblen45) [inline typescript/src/data_models/manual_task/manual_task.ts]: process owner should be better

File: typescript/src/data_models/messages/bpmn_events/base_bpmn_event_message.ts [hunk 2]
```diff
@@ -1,19 +1,32 @@
...
-  public processModelId: string;
-  public processInstanceId: string;
-  public flowNodeId: string;
-  public currentToken: any;
+  public readonly correlationId: string;
+  public readonly processModelId: string;
+  public readonly processInstanceId: string;
+  public readonly flowNodeId: string;
+  public readonly flowNodeInstanceId: string;
+  public readonly identity: IIdentity;
```
Comment (roblen45) [inline typescript/src/data_models/messages/bpmn_events/base_bpmn_event_message.ts]: same here and in every message

File: typescript/src/data_models/user_task/user_task.ts [hunk 3]
```diff
@@ -36,4 +36,8 @@ export class UserTask {
    * The token payload the UserTask got suspended with.
    */
   public tokenPayload: any;
+  /**
+   * Contains the identity the UserTask belongs to.
+   */
+  public identity: any;
```
Comment (roblen45) [inline typescript/src/data_models/user_task/user_task.ts]: process owner

**Comment by ElRaptorus** [comment]
@roblen45 done

File: typescript/src/data_models/manual_task/manual_task.ts [hunk 4]
```diff
@@ -30,4 +32,8 @@ export class ManualTask {
    * The token payload the ManualTask got suspended with.
    */
   public tokenPayload: any;
+  /**
+   * Contains the identity the UserTask belongs to.
+   */
+  public owner: IIdentity;
```
Comment (roblen45) [inline typescript/src/data_models/manual_task/manual_task.ts]: processInstanceOwner

File: typescript/src/data_models/user_task/user_task.ts [hunk 5]
```diff
@@ -36,4 +38,8 @@ export class UserTask {
    * The token payload the UserTask got suspended with.
    */
   public tokenPayload: any;
+  /**
+   * Contains the identity the UserTask belongs to.
+   */
+  public owner: IIdentity;
```
Comment (roblen45) [inline typescript/src/data_models/user_task/user_task.ts]: processInstanceOwner

**Comment by ElRaptorus** [comment]
@roblen45 done

**Pull request merged**